src/main/com/bazelbuild/java/classpath/ClassPathValidator.java

```diff
@@ -62,15 +66,29 @@ private ClasspathCollision collisionsBetween(Map<String, Map<String, String>> ta
             Map<String, String> classpathToDigest = new HashMap<>();
             jarEntries.stream()
                     .filter(g -> !ignored(g.path))
+                    .filter(g -> included(g.path))
                     .forEach(e-> classpathToDigest.put(e.getPath(),e.getDigest()));
```

Comment:
WDYT abiyt adding the `included` filter only iff `(this.includePrefixes.isEmpty() && this.includeSuffixes.isEmpty())`... If we do it in this level this would become a little more effecient

Reply:
Done!